config.json

```diff
@@ -2,87 +2,85 @@
-            "namespace": "Azure::ARM::Batch",
-            "version": "0.11.0",
+            "namespace": "Azure::ARM::Batch::Api_2017_01_01",
+            "package-version": "0.11.0",
             "output-folder": "2017-01-01"
         },
         "azure_mgmt_batch_2015_12_01":{
             "input-file": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/batch/resource-manager/Microsoft.Batch/2015-12-01/BatchManagement.json",
-            "namespace": "Azure::ARM::Batch",
-            "version": "0.11.0",
+            "namespace": "Azure::ARM::Batch::Api_2015_12_01",
+            "package-version": "0.11.0",
             "output-folder": "2015-12-01"
         }
     },
     "azure_mgmt_cdn": {
         "azure_mgmt_cdn_2016_10_02":{
             "input-file": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/cdn/resource-manager/Microsoft.Cdn/2016-10-02/cdn.json",
-            "namespace": "Azure::ARM::CDN",
-            "version": "0.11.0",
-            "output-folder": "2016-10-02"
+            "namespace": "Azure::ARM::CDN::Api_2016_10_02",
+            "package-version": "0.11.0"
         },
         "azure_mgmt_cdn_2016_04_02":{
             "input-file": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/cdn/resource-manager/Microsoft.Cdn/2016-04-02/cdn.json",
-            "namespace": "Azure::ARM::CDN",
-            "version": "0.11.0",
-            "output-folder": "2016-04-02"
+            "namespace": "Azure::ARM::CDN::Api_2016_04_02",
+            "package-version": "0.11.0"
         },
         "azure_mgmt_cdn_2015_06_01":{
             "input-file": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/cdn/resource-manager/Microsoft.Cdn/2015-06-01/cdn.json",
-            "namespace": "Azure::ARM::Batch",
-            "version": "0.11.0",
+            "namespace": "Azure::ARM::Batch::Api_2015_06_01",
+            "package-version": "0.11.0",
             "output-folder": "2015-06-01"
         }
     },
     "azure_mgmt_compute":{
         "azure_mgmt_compute_2017_03": {
             "markdown": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/compute/resource-manager/readme.md",
-            "namespace": "Azure::ARM::Compute",
-            "version": "0.11.0",
-            "output-folder": "2017-03-30",
+            "namespace": "Azure::ARM::Compute::Api_2017_03_30",
+            "package-version": "0.11.0",
             "tag": "package-compute-2017-03"
         },
         "azure_mgmt_compute_2016_04_preview": {
             "markdown": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/compute/resource-manager/readme.md",
-            "namespace": "Azure::ARM::Compute",
-            "version": "0.11.0",
-            "output-folder": "2016-04-preview",
+            "namespace": "Azure::ARM::Compute::Api_2016_04_preview",
```

Comment:
Curious: Here we are defining `Azure::ARM::Compute::Api_xxxx_xx_preview` that is different from api version value `2016-04-30-preview` is that expected? That seems folder name in generated code is different from namespace. I am bit confused here whether this is expected or not.